### ISSUE TITLE
encryption that is compatible with Java AES/CFB/PKCS5Padding (padding issue?)

### ISSUE THREAD

**Issue by MartijnR** [digitalbazaar/forge#612]
Hi,

Thanks very much for this library!

I'm trying to encrypt content that is later decrypted in Java with AES/CFB/PKCS5Padding (the latter is 
fixed, part of an existing application).

The error during Java decryption is:
```
Caused by: javax.crypto.IllegalBlockSizeException: last block incomplete in decryption
```

(decryption in javascript works fine)

The key length is 32 bytes.
The iv length is 16 bytes.

This is my code:

```js
    var SYMMETRIC_ALGORITHM = 'AES-CFB';
    var cipher = forge.cipher.createCipher( SYMMETRIC_ALGORITHM, symmetricKey );
   
    cipher.start( {
        iv: ivSeedArray.join( '' )
    } );
    cipher.update( forge.util.createBuffer( content ) );

    // manual padding: https://github.com/digitalbazaar/forge/issues/100#issuecomment-34837467
    var pass = cipher.finish( function( blockSize, buffer ) {
        var padding = blockSize - ( buffer.length() % blockSize );
        buffer.fillWithByte( padding, padding );
        return true;
    } );

    var byteString = cipher.output.getBytes();

    // write the bytes of the string to an ArrayBuffer
    var buffer = new ArrayBuffer( byteString.length );
    var array = new Uint8Array( buffer );

    for ( var i = 0; i < byteString.length; i++ ) {
        array[ i ] = byteString.charCodeAt( i );
    }

    // write the ArrayBuffer to a blob
    var blob = new Blob( [ array ] );
```

I've read padding doesn't make much sense with AES-CFB, but we need to match Java in this case. I suspect the padding is where it goes wrong based on the Java error message.

Any ideas on what I am doing wrong?

**Comment by MartijnR** [comment]
After looking at the source code, I realized, the padding function passed `cipher.finish()` doesn't actually do anything for 'CFB' mode.

So I switched to a lower level padding replacement function:

```
cipher.mode.pad = function( input ) {
        var padding = this.blockSize - ( input.length() % this.blockSize );
        input.fillWithByte( padding, padding );
        return true;
    };
cipher.finish()
```

Now, it looks like I'm nearly there as it is decrypting in Java. However, there is a parsing error of the decrypted XML. I think the start of the output is garbled up. I'm guessing my padding function and Java's AES/CFB/PKCS5Padding unpadding function are not compatible.

Any suggestions on how to tweak the padding function?
